### ISSUE TITLE
TreeCluster : is it only for binary trees ?

### ISSUE THREAD

**Issue by ghsama** [niemasd/TreeCluster#2]
Hey,
I was wondering if the algorithm is only usable for binary trees, and if so is there a possibility to generalize it to all kind of trees (or at least rooted ones).
Thanks

**Comment by niemasd** [comment]
It should work on arbitrary trees (not necessarily binary)

The methods that don't say "Clare" should work on both rooted and unrooted trees if I'm not mistaken

**Comment by niemasd** [comment]
I'm going to close this issue because it seems like the question is answered, but please feel free to follow up (either on this Issue or in a new one) and I'll be happy to help :-)

**Comment by ghsama** [comment]
Hey,
thank you for you response.But in the original article (https://journals.plos.org/plosone/article?id=10.1371/journal.pone.0221068) it states in the problem definition `Let T = (V, E) be an unrooted binary tree represented by an undirected acyclic graph with vertices V (each with degree one or three), weighted edges E, and leafset` . andd also in the algorithm 1, it's based on the fact that there is 'left chld' and 'right child'. If you came up to have more g eneral algo which is the implemented, could you please give me a reference. thank you

**Comment by niemasd** [comment]
Given a tree with polytomies, we can simply resolve all polytomies arbitrarily with 0-length branches to yield a binary tree that is equivalent with respect to the clustering definitions we provide. For example, imagine the following tree:

```
(A:1,B:2,C:3);

--- A
|
------ B
|
--------- C

--- = length of 1
```

If we arbitrarily resolve the polytomy with 0-length branches, we could get the following binary tree:

```
((A:1,B:2):0,C:3);

  --- A
  |
|-
| |
| ------ B
|
--------- C

--- = length of 1
-   = length of 0
```

As can be seen, the addition of this 0-length branch does not impact pairwise distances, root-to-tip distances, maximum branch length, etc.

**Comment by niemasd** [comment]
I'll close this issue because it seems like the question is answered, but please feel free to follow up (either on this Issue or in a new one) and I'll be happy to help!

TL;DR: Given a non-binary tree, TreeCluster resolves polytomies arbitrarily with 0-length branches, which does not impact any of the clustering methods implemented in TreeCluster
